src/main/kotlin/solve/utils/materialfx/MFXNodesUtils.kt

```diff
@@ -0,0 +1,128 @@
+package solve.utils.materialfx
+
+import io.github.palexdev.materialfx.controls.MFXButton
+import io.github.palexdev.materialfx.controls.MFXCheckListView
+import io.github.palexdev.materialfx.controls.MFXCheckbox
+import io.github.palexdev.materialfx.controls.MFXContextMenu
+import io.github.palexdev.materialfx.controls.MFXContextMenuItem
+import io.github.palexdev.materialfx.controls.MFXTextField
+import io.github.palexdev.materialfx.effects.ripple.RippleClipType
+import io.github.palexdev.materialfx.factories.RippleClipTypeFactory
+import javafx.collections.ObservableList
+import javafx.event.EventTarget
+import javafx.scene.Node
+import javafx.scene.paint.Color
+import org.controlsfx.control.RangeSlider
+import solve.utils.materialfx.stylesheets.MFXRangeSliderStylesheet
+import tornadofx.*
+import java.util.function.Supplier
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToLong
+
+fun EventTarget.mfxButton(
+    text: String = "",
+    graphic: Node? = null,
+    op: MFXButton.() -> Unit = {}
+) = MFXButton(text).attachTo(this, op) {
+    if (graphic != null) it.graphic = graphic
+}
+
+fun EventTarget.mfxTextField(
+    text: String = "",
+    op: MFXTextField.() -> Unit = {}
+) = MFXTextField(text).attachTo(this, op)
+
+fun Node.mfxContextMenu(op: MFXContextMenu.() -> Unit = {}): MFXContextMenu {
+    val contextMenu = MFXContextMenu.Builder.build(this).installAndGet()
+    op(contextMenu)
+    return contextMenu
+}
+
+fun MFXContextMenu.item(
+    name: String,
+    op: MFXContextMenuItem.() -> Unit = {}
+) = MFXContextMenuItem(name).also {
+    op(it)
+    this.addItems(it)
+}
+
+fun MFXContextMenu.lineSeparator() = this.addLineSeparator(MFXContextMenu.Builder.getLineSeparator())
+
+fun MFXContextMenuItem.action(op: () -> Unit) = this.setOnAction { op() }
+
+fun Node.mfxCircleButton(
+    graphic: Node? = null,
+    radius: Double = 20.0,
+    text: String = "",
+    op: MFXButton.() -> Unit = {}
+) = mfxButton(text, graphic) {
+    this.graphic = graphic
+    rippleGenerator.clipSupplier = Supplier {
+        RippleClipTypeFactory(RippleClipType.CIRCLE).setRadius(radius).build(this)
+    }
+    style {
```

Comment:
I think it's better to leave it inside, because we have a radius parameter passed as an argument, which is needed for both the button and the ripple factory when button is pressed. It also ensures that no other set of styles will change the settings of the button created in this function.